letsencrypt/cli.py

```diff
@@ -695,6 +725,168 @@ def install(config, plugins):
     le_client.enhance_config(domains, config)
 
 
+def _restore_required_config_elements(full_path, config, renewalparams):
+    # string-valued items to add if they're present
+    for config_item in STR_CONFIG_ITEMS:
+        if config_item in renewalparams:
+            value = renewalparams[config_item]
+            # Unfortunately, we've lost type information from ConfigObj,
+            # so we don't know if the original was NoneType or str!
+            if value == "None":
+                value = None
+            setattr(config.namespace, config_item, value)
+    # int-valued items to add if they're present
+    for config_item in INT_CONFIG_ITEMS:
+        if config_item in renewalparams:
+            try:
+                value = int(renewalparams[config_item])
+                setattr(config.namespace, config_item, value)
+            except ValueError:
+                logger.warning("Renewal configuration file %s specifies "
+                               "a non-numeric value for %s. Skipping.",
+                               full_path, config_item)
+                raise
+
+def _restore_plugin_configs(config, renewalparams):
+    # Now use parser to get plugin-prefixed items with correct types
+    # XXX: the current approach of extracting only prefixed items
+    #      related to the actually-used installer and authenticator
+    #      works as long as plugins don't need to read plugin-specific
+    #      variables set by someone else (e.g., assuming Apache
+    #      configurator doesn't need to read webroot_ variables).
+    # XXX: is it true that an item will end up in _parser._actions even
+    #      when no action was explicitly specified?
+    plugin_prefixes = [renewalparams["authenticator"]]
+    if "installer" in renewalparams and renewalparams["installer"] != None:
+        plugin_prefixes.append(renewalparams["installer"])
+    for plugin_prefix in set(renewalparams):
+        for config_item in renewalparams.keys():
+            if renewalparams[config_item] == "None":
+                # Avoid confusion when, for example, "csr = None" (avoid
+                # trying to read the file called "None")
+                # Should we omit the item entirely rather than setting
+                # its value to None?
+                setattr(config.namespace, config_item, None)
+                continue
+            if config_item.startswith(plugin_prefix + "_"):
+                for action in _parser.parser._actions: # pylint: disable=protected-access
```

Comment:
This will have to be changed if the behavior of `None` values is inappropriate, but this `for` loop and the accompanying `else` can be replaced with these two lines of code:
```
typ = next((a.type for a in _parser.parser._actions
            if a.type is not None and a.dest == config_item), str)
setattr(config.namespace, config_item, typ(renewalparams[config_item]))
```